### PR TITLE
104 x dev nano

### PR DESCRIPTION
**Pull request by sscruz**

### CODE REVIEW

**Comment by gpetruc** [comment]
Looks ok to me.
@peruzzim do you want to have a look before I merge?

File: TTHAnalysis/python/plotter/ttH-multilepton/2lss_3l_plots.txt [hunk 1]
```diff
@@ -192,68 +192,68 @@ lep1_phi: 	 LepGood1_phi					 : 20,-3.14,3.14; XTitle="Leading lepton #phi"
...
+lep1_miniIsoNeu: LepGood1_miniPFRelIso_all-LepGood1_miniPFRelIso_chg : 40,0,0.4; XTitle="Leading lepton neutral miniIso", Logy
 lep1_sip3d:	 LepGood1_sip3d					 : 40,0,8; XTitle="Leading lepton S_{IP} 3D"
 lep1_dxy:	 abs(LepGood1_dxy)*10000			 : 40,0.0,100.; XTitle="Leading lepton dxy [#mum]"
 lep1_dz:	 abs(LepGood1_dz)*10000			 	 : 40,0.0,100.; XTitle="Leading lepton dz [#mum]"
-lep1_ptratio:	 LepGood1_jetPtRatiov2				 : 40,0,1.2; XTitle="Leading lepton ptRatio", Legend='TL'
+lep1_ptratio:	 LepGood1_jetRelIso				 : 40,0,1.2; XTitle="Leading lepton jetRelIso", Legend='TL'
 lep1_ptrel:	 LepGood1_jetPtRelv2				 : 40,0,40; XTitle="Leading lepton ptRel", Legend='TL'
-lep1_jetBTagCSV: LepGood1_jetBTagCSV				 : 40,0,1; XTitle="Leading lepton CSV of nearest jet"
-lep1_segcomp:	 if3(abs(LepGood1_pdgId)==13,LepGood1_segmentCompatibility,-99) : 40,0,1; XTitle="Leading lepton segment compat.", Legend='TL', IncludeOverflows=False
+lep1_jetBTagCSV: LepGood1_jetBTagDeepFlav				 : 40,0,1; XTitle="Leading lepton deep jet of nearest jet"
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/ttH-multilepton/2lss_3l_plots.txt]: could you please adapt the name to lep1_jetBTagDeepFlav?

File: TTHAnalysis/python/plotter/ttH-multilepton/2lss_3l_plots.txt [hunk 2]
```diff
@@ -192,68 +192,68 @@ lep1_phi: 	 LepGood1_phi					 : 20,-3.14,3.14; XTitle="Leading lepton #phi"
...
+lep1_elIdMVA:	 -0.05*log((1-LepGood1_mvaFall17V2noIso)/(1+LepGood1_mvaFall17V2noIso))				 : 40,0,1; XTitle="Leading lepton electron ID MVA", Legend='TL', Logy
 lep1_relIso04:	 LepGood1_relIso04				 : 40,0,0.5; XTitle="Leading lepton relIso04", Legend='TR', Logy
 lep1_relIso03:	 LepGood1_relIso03				 : 40,0,0.5; XTitle="Leading lepton relIso03", Legend='TR', Logy
 
 lep2_pt: 	 LepGood2_pt					 : 40,0,200; XTitle="Sub-leading lepton p_{T} (GeV)"
 lep2_conePt: 	 LepGood2_conePt				 : 40,0,100; XTitle="Sub-leading lepton cone p_{T} (GeV)"
 lep2_eta: 	 LepGood2_eta					 : 40,-2.5,2.5; XTitle="Sub-leading lepton #eta"
 lep2_phi: 	 LepGood2_phi					 : 20,-3.14,3.14; XTitle="Sub-leading lepton #phi"
-lep2_pdgId:	 0.5*LepGood2_pdgId/abs(LepGood2_pdgId)*(abs(LepGood2_pdgId)-10) : [-2,-1,0,1,2]; XTitle="Sub-leading lepton PDG ID (#mu^{-}/e^{-}/e^{+}/#mu^{+})", XNDdiv=505
+lep2_pdgId:	 -0.05*LepGood2_pdgId/abs(LepGood2_pdgId)*(abs(LepGood2_pdgId)-10) : [-2,-1,0,1,2]; XTitle="Sub-leading lepton PDG ID (#mu^{-}/e^{-}/e^{+}/#mu^{+})", XNDdiv=505
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/ttH-multilepton/2lss_3l_plots.txt]: why this 0.05?

File: TTHAnalysis/python/plotter/ttH-multilepton/lepchoice-ttH-FO.txt [hunk 3]
```diff
@@ -6,6 +6,13 @@ LepGood1_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[0]]==1 || LepGo
...
 LepGood3_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[2]]==1 || LepGood_genPartFlav[iLepFO_Recl[2]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[2]]]*LepGood_pdgId[iLepFO_Recl[2]]>0); AlsoData
 LepGood4_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[3]]==1 || LepGood_genPartFlav[iLepFO_Recl[3]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[3]]]*LepGood_pdgId[iLepFO_Recl[3]]>0); AlsoData
+LepGood1_mcMatchId\b : (LepGood_genPartFlav[iLepFO_Recl[0]]==1 || LepGood_genPartFlav[iLepFO_Recl[0]]==15); AlsoData
+LepGood2_mcMatchId\b : (LepGood_genPartFlav[iLepFO_Recl[1]]==1 || LepGood_genPartFlav[iLepFO_Recl[1]]==15); AlsoData
+LepGood3_mcMatchId\b : (LepGood_genPartFlav[iLepFO_Recl[2]]==1 || LepGood_genPartFlav[iLepFO_Recl[2]]==15); AlsoData
+LepGood4_mcMatchId\b : (LepGood_genPartFlav[iLepFO_Recl[3]]==1 || LepGood_genPartFlav[iLepFO_Recl[3]]==15); AlsoData
+LepGood1_mcPromptGamma\b  : (LepGood_genPartFlav[0] == 22); AlsoData
+LepGood2_mcPromptGamma\b  : (LepGood_genPartFlav[1] == 22); AlsoData
+LepGood3_mcPromptGamma\b  : (LepGood_genPartFlav[2] == 22); AlsoData
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/ttH-multilepton/lepchoice-ttH-FO.txt:15]: please also add LepGood4_mcPromptGamma for consistency/completeness

File: TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py [hunk 4]
```diff
@@ -315,6 +320,11 @@ def doprescale3l(x,torun):
         x = base('2lss')
         x = x.replace('mca-2lss-mc.txt','mca-2lss-mcdata-ttbar.txt')
         x = x.replace('--maxRatioRange 0 3','--maxRatioRange 0.8 1.2')
+        x = x.replace('--rebin 4','')
+        if '_appl' in torun:
+            x = add(x,"-X TT")
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py]: more prudent to add ^ in front, as for many other exclude/invert/enable commands

**Review by peruzzim — CHANGES_REQUESTED** [review]

**Pull request merged**